### ISSUE TITLE
Which labels to use with transformers libary ?

### ISSUE THREAD

**Issue by LarsAC** [EmilyAlsentzer/clinicalBERT#29]
Hello,

this looks like a great piece of work, thank you for making it available. I tried to explore clinicalBERT for some NER tasks using the transformers library. I can obtain a list of token index results from torch.argmax, but I cannot find a suitable a set of labels (`predictions` is containing values as large as 687). What am I doing  wrong ?

```
from transformers import AutoTokenizer, AutoModel
import torch

tokenizer = AutoTokenizer.from_pretrained("emilyalsentzer/Bio_ClinicalBERT")

model = AutoModel.from_pretrained("emilyalsentzer/Bio_ClinicalBERT")

label_list = ["B-IDNUM", "I-IDNUM", "B-HOSPITAL", "I-HOSPITAL", 'B-PATIENT', 'I-PATIENT', 'B-PHONE', 'I-PHONE',
        'B-DATE', 'I-DATE', 'B-DOCTOR', 'I-DOCTOR', 'B-LOCATION-OTHER', 'I-LOCATION-OTHER', 'B-AGE', 'I-AGE', 'B-BIOID', 'I-BIOID',
        'B-STATE', 'I-STATE','B-ZIP', 'I-ZIP', 'B-HEALTHPLAN', 'I-HEALTHPLAN', 'B-ORGANIZATION', 'I-ORGANIZATION',
        'B-MEDICALRECORD', 'I-MEDICALRECORD', 'B-CITY', 'I-CITY', 'B-STREET', 'I-STREET', 'B-COUNTRY', 'I-COUNTRY',
        'B-URL', 'I-URL',
        'B-USERNAME', 'I-USERNAME', 'B-PROFESSION', 'I-PROFESSION', 'B-FAX', 'I-FAX', 'B-EMAIL', 'I-EMAIL', 'B-DEVICE', 'I-DEVICE',
        'O', "X", "[CLS]", "[SEP]"]

sequence = "Patient had severe headache and took two Aspirine."

# Bit of a hack to get the tokens with the special tokens
tokens = tokenizer.tokenize(tokenizer.decode(tokenizer.encode(sequence)))
inputs = tokenizer.encode(sequence, return_tensors="pt")

outputs = model(inputs)[0]
predictions = torch.argmax(outputs, dim=2)

print([(token, label_list[prediction]) for token, prediction in zip(tokens, predictions[0].tolist())])
```

Lars

**Comment by vatsal-sodha** [comment]
Hi,
I have run into a similar kind of issue. As this repository only provided clinical BERT models that were pretrained via masked LM and NSP on MIMIC data but did not release any of the models that we fine-tuned on the i2b2 or MedNLI tasks. #22 

The easy workaround for NER tasks can be as follows:
```
from transformers import AutoTokenizer, AutoModelForTokenClassification
import torch

tokenizer = AutoTokenizer.from_pretrained("emilyalsentzer/Bio_ClinicalBERT")
model = AutoModelForTokenClassification.from_pretrained("emilyalsentzer/Bio_ClinicalBERT")
```

AutoModelForTokenClassification will append a token classification head and will import weights for the BERT model. Don't forget to fine-tune this model.

Let me know if it helped you.
-SodhA

**Comment by EmilyAlsentzer** [comment]
Hi Lars,

@vatsal-sodha is correct. We're unfortunately unable to upload the NER models trained on i2b2 to huggingface because we trained these in TF and the conversion scripts for converting TF to Pytorch models only work for the pretrained models as far as I understand. Your best bet is to do what @vatsal-sodha describes. Load in the clinicalBERT base and train the model on the downstream dataset of your choice. If you do this, let me know and we can feature it on this repo.

Closing this issue now, but feel free to reopen or open a new one if you have additional questions.
